File: src/core/format/localeConfig.ts
export interface LocaleConfig {
  readonly numberLocale: string;
  readonly dateLocale: string;
  readonly currencySymbol: string;
}

export const localeConfigs: Readonly<Record<string, LocaleConfig>> = {
  en: { numberLocale: 'en-IN', dateLocale: 'en', currencySymbol: '₹' },
  hi: { numberLocale: 'hi', dateLocale: 'hi', currencySymbol: '₹' },
  ar: { numberLocale: 'ar-EG', dateLocale: 'ar-EG', currencySymbol: '₹' },
};

export const supportedLanguages = ['en', 'hi', 'ar'] as const;
export type SupportedLanguage = (typeof supportedLanguages)[number];

File: lib/core/format/locale_config.dart
class LocaleConfig {
  const LocaleConfig({
    required this.numberLocale,
    required this.dateLocale,
    required this.currencySymbol,
  });

  final String numberLocale;
  final String dateLocale;
  final String currencySymbol;
}

const localeConfigs = <String, LocaleConfig>{
  'en': LocaleConfig(numberLocale: 'en_IN', dateLocale: 'en', currencySymbol: '₹'),
  'hi': LocaleConfig(numberLocale: 'hi', dateLocale: 'hi', currencySymbol: '₹'),
  'ar': LocaleConfig(
      numberLocale: 'ar_EG', dateLocale: 'ar_EG', currencySymbol: '₹'),
};
